index.js

```diff
@@ -357,7 +360,53 @@ function remoteFetch (uri, opts) {
             return retryHandler(res)
           }
 
-          return res
+          if (!fetch.isRedirect(res.status) || opts.redirect === 'manual') {
+            return res
+          }
+
+          // handle redirects - matches behavior of npm-fetch: https://github.com/bitinn/node-fetch
+          if (opts.redirect === 'error') {
+            throw new Error(`redirect mode is set to error: ${uri}`, 'no-redirect')
+          }
+
+          if (!res.headers.get('location')) {
+            throw new Error(`redirect location header missing at: ${uri}`, 'invalid-redirect')
+          }
+
+          if (req.counter >= req.follow) {
+            throw new Error(`maximum redirect reached at: ${uri}`, 'max-redirect')
```

Comment:
What does this second argument do? The convention so far has been to use a `code` property on error objects to have single codes that people can use.